### PR TITLE
FT-90 FT-8 feat: theme style ids

### PR DESCRIPTION
**Pull request by LiamMartens**

### CODE REVIEW

**Comment by LiamMartens** [comment]
@six7 I added the theme prefix setting and updated the logic to support remote style lookup

File: src/figmaStorage/UiSettingsProperty.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { SavedSettings } from '@/plugin/notifiers';
+import { tryParseJson } from '@/utils/tryParseJson';
+import { FigmaStorageProperty, FigmaStorageType } from './FigmaStorageProperty';
+
+export const UiSettingsProperty = new FigmaStorageProperty<Partial<SavedSettings>>(
+  FigmaStorageType.CLIENT_STORAGE,
+  'apiProviders',
```
Comment (six7) [inline src/figmaStorage/UiSettingsProperty.ts:7]: heads up: this just caused me to lose all my locally stored api providers as you're overwriting the wrong property here :D

**Comment by six7** [comment]
Functionally this works fine now! 🎉 Can we break it down in smaller PRs so we have most of the code in without a big bang?

**Comment by LiamMartens** [comment]
@six7 is that worth it? There are 59 changed files excluding the updated test files

**Comment by six7** [comment]
> @six7 is that worth it? There are 59 changed files excluding the updated test files

Not necessarily, we can keep testing this branch to make sure it's all good to go, but thought it might make sense to merge parts of it in earlier to reduce merge conflict potential.

**Comment by six7** [comment]
> @six7 this was tested by esther and is ready

yes, we're running it as a beta for now, meaning we sent users a .zip with the build. i'll do a release later today that fixes some issues (not concerning this pr), after that release i'll merge this one into the `next` branch.

**Comment by six7** [comment]
> @six7 are we getting this through?

See Slack on that one

**Review by six7 — APPROVED** [review]

**Pull request closed without merging**